homeassistant/components/feedreader.py

```diff
@@ -0,0 +1,96 @@
+        """Initialize the FeedManager object."""
+        self._url = url
+        self._feed = None
+        self._hass = hass
+        # Initialize last entry timestamp as epoch time
+        self._last_entry_timestamp = datetime.utcfromtimestamp(0).timetuple()
+        _LOGGER.debug('Loading feed %s', self._url)
+
+    @property
+    def url(self):
+        """Url of the feed."""
+        return self._url
+
+    @property
+    def etag(self):
+        """Etag of the latest feed update if available."""
+        return None if not self._feed else self._feed.get('etag')
+
+    @property
+    def lastmodified_header(self):
+        """Last-Modified header of the latest feed update if available."""
+        return None if not self._feed else self._feed.get('modified')
+
+    @property
+    def errors(self):
+        """Return True if there was any error parsing feed."""
+        return True if not self._feed else self._feed.bozo != 0
+
+    def _log_no_entries(self):
+        """Send no entries log at debug level."""
+        _LOGGER.debug('No new entries in feed %s', self.url)
+
+    def update(self):
+        """Update the feed and publish new entries in the event bus."""
+        import feedparser
+        _LOGGER.info('Fetching new data from feed %s', self.url)
+        self._feed = feedparser.parse(self.url,
+                                      etag=self.etag,
+                                      modified=self.lastmodified_header)
+        if self.errors:
+            _LOGGER.error('Error parsing feed %s', self.url)
+        # Using etag and modified, if there's no new data available,
+        # the entries list will be empty
+        elif len(self._feed.entries) > 0:
+            self._publish_new_entries()
+            self._last_entry_timestamp = self._feed.entries[0].published_parsed
+        else:
+            self._log_no_entries()
+
+    def _publish_new_entries(self):
+        """Publish new entries to the event bus."""
+        new_entries = False
+        for entry in self._feed.entries:
+            # Consider only entries newer then the latest parsed one
+            if entry.published_parsed > self._last_entry_timestamp:
+                new_entries = True
+                entry.update({'feed_url': self.url})
+                self._hass.bus.fire(EVENT_FEEDREADER, entry)
+        if not new_entries:
+            self._log_no_entries()
+
+
+def setup(hass, config):
+    """Setup the feedreader component."""
+    if not validate_config(config, {DOMAIN: ['urls']}, _LOGGER):
+        _LOGGER.error("You must define a list of URL addresses "
+                      "for your feedreader in the config file. "
+                      "Setup aborted")
+        return False
+    urls = config.get(DOMAIN)['urls']
+    feeds = [FeedManager(url, hass) for url in urls]
+    for feed in feeds:
+        # Update feeds every hour
+        track_utc_time_change(hass, lambda now: feed.update(), minute=0, second=0)
```

Comment:
Never mind, I think I found out already, https://github.com/home-assistant/home-assistant/pull/1836/commits/dae4318bf2957fe251b5093fefcafd66f4ca7252 should fix